File: ui/src/app/business/deploy-plan/ip-pool/ip/ip.ts
import {BaseModel, BaseRequest} from '../../../../shared/class/BaseModel';

export class Ip extends BaseModel {
    address: string;
    gateway: string;
    dns1: string;
    dns2: string;
    status: string;
}

export class IpCreate extends BaseRequest {
    ipStart: string;
    ipEnd: string;
    gateway: string;
    dns1: string;
    dns2: string;
    ipPoolName: string;
    subnet: string;
}

export class IpUpdate extends BaseRequest {
    address: string;
    operation: string;
}

File: pkg/dto/ip.go
package dto

import "github.com/KubeOperator/KubeOperator/pkg/model"

type Ip struct {
	model.Ip
	IpPool model.IpPool `json:"ipPool"`
}

type IpCreate struct {
	IpStart    string `json:"ipStart" validate:"required,koip"`
	IpEnd      string `json:"ipEnd" validate:"required,koip"`
	Subnet     string `json:"subnet"`
	Gateway    string `json:"gateway"`
	DNS1       string `json:"dns1"`
	DNS2       string `json:"dns2"`
	IpPoolName string `json:"ipPoolName" validate:"required"`
}

type IpOp struct {
	Operation string `json:"operation"  validate:"required"`
	Items     []Ip   `json:"items"  validate:"required"`
}

type IpUpdate struct {
	Address   string `json:"address"`
	Operation string `json:"operation"`
}

type IpSync struct {
	IpPoolName string `json:"ipPoolName"`
}
